src/cli/self_update.rs

```diff
@@ -1125,8 +1125,34 @@ pub(crate) fn self_update_permitted(explicit: bool) -> Result<SelfUpdatePermissi
     Ok(SelfUpdatePermission::Permit)
 }
 
-/// Performs all of a self-update: check policy, download, apply and exit.
-pub(crate) async fn self_update(dl_cfg: &DownloadCfg<'_>) -> Result<ExitCode> {
+/// Optionally performs a self-update: check policy, download, apply and exit.
+///
+/// Whether the self-update is executed is based on both compile-time and runtime
+/// configurations, where the priority is as follows:
+/// no-self-update feature > self update mode > no-self-update args.
+///
+/// i.e. update only if rustup does **not** have the no-self-update feature,
+/// and self update mode is configured to **enable**
+/// and has **no** no-self-update parameter.
+pub(crate) async fn self_update(
+    dl_cfg: &DownloadCfg<'_>,
```

Comment:
Nit: suggest keeping `dl_cfg` last (since it's a less specific argument for this function).

Consider making this a `SelfUpdateMethod` method, maybe `update()` or `run()` or `execute()`?